### PR TITLE
127 log classification metrics to wandb

### PR DESCRIPTION
**Pull request by kapiblue**
This PR aims to prepare the classification experiment:

1. Log classification metrics to wandb
2. Save `pos_weights` to file
3. Enable finetuning from classification to regression

### CODE REVIEW

**Comment by kapiblue** [comment]
TODO: Enable finetuning from classification to regression

Use the same total number of epochs in the experiment. For example, 45 epochs for clf, and 5 epochs for reg.

**Comment by kapiblue** [comment]
### Finetuning

Finetuning enables moving from the classification scenario to regression.

#### Step 1: Train the model for classification

```bash
python .\src\train.py hydra.run.dir=results\test_wandb_clf training.epochs=5  data.is_classification=True    
```

#### Step 2: Finetune

```bash
python .\src\finetune.py -cp ../results/test_wandb_clf hydra.run.dir=. hydra.output_subdir=null hydra/job_logging=disabled hydra/hydra_logging=disabled training.epochs=5
```

This script will load the best classification model, run it on the test set, create the classification report and save it.

Next, it finetunes the model for regression. For now, it uses mse loss function and sigmoid activation of the predictor:

```python

loss_fn_name = "mse"

# Modify the predictor activation function
    model.predictor.activation = "sigmoid"
```

File names and logging paths were changed to reflect the scenario, for example, "ft_best.pth".

#### Step 3: Test regression

Testing script remains the same. However, we need to override the loss function and activation:

```bash
.\src\test.py -cp ../results/test_wandb_clf hydra.run.dir=. hydra.output_subdir=null hydra/job_logging=disabled hydra/hydra_logging=disabled testing.weights=ft_best.pth training.loss_function=mse training.predictor.activation=relu data.is_classification=False
```
